frontend/pages/dashboard/projects.tsx

```diff
@@ -81,7 +81,10 @@ export const ProjectsPage: PageComponent<ProjectsPageProps, DashboardLayoutProps
   const handleRowMenuItemClick = ({ key, slug }: { key: string; slug: string }) => {
     switch (key) {
       case 'edit':
-        router.push(`${Paths.Project}/${slug}/edit`);
+        router.push(
+          `${Paths.Project}/${slug}/edit?returnPath=${encodeURIComponent(router.asPath)}`,
```

Comment:
The URLs in this particular view can be something like: 

`/dashboard/projects?page=1&search=&sorting=category+desc`

While passing that as a `returnPath` param, everything after the first ampersand would be stripped. By encoding this we preserve the full return path.